REGRESSION(274325@main): didFindString is always false when searching in PDFs

charliewolfe opened the pull request:
#### 2db6785ed8fff72f4235a477450e2efbc555197d
<pre>
REGRESSION(274325@main): didFindString is always false when searching in PDFs
<a href="https://rdar.apple.com/122901465">rdar://122901465</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=269310">https://bugs.webkit.org/show_bug.cgi?id=269310</a>

Reviewed by Sammy Gill.

After 274325@main, API::FindClient::didFindString() is called in the completion handler of the final
FindString message if a match was found inside a frame. For PDFs, this frame ID was never set, so
didFindString would never be called. For now, if a match is found in a PDF, mark the main frame as the
frame containing the match.

* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):

Canonical link: <a href="https://commits.webkit.org/274585@main">https://commits.webkit.org/274585@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f8326cbb10eb5ae31bd384f4851599b8fffd711f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39493 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18472 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41847 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42027 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35393 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21353 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15801 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/33003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40067 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15560 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/34201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/13515 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43305 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35853 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35481 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39278 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11791 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37532 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15911 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/15959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5178 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/WebPage/FindController.cpp:
```diff
@@ -285,9 +285,10 @@ void FindController::findString(const String& string, OptionSet<FindOptions> opt
-    else
+    if (pluginView) {
+        if ((found = pluginView->findString(string, coreOptions, maxMatchCount)))
+            idOfFrameContainingString = m_webPage->corePage()->mainFrame().frameID();
```
Mind adding a FIXME here for the iframe case?

aprotyas replied:
Also, `PluginView` has access to a `Frame` object. Would that contain the right frame ID?

charliewolfe replied:
Looks like it, I'll use that frame instead.

aprotyas reviewed (commented):
LGTM

charliewolfe committed:
REGRESSION(274325@main): didFindString is always false when searching in PDFs
rdar://122901465
https://bugs.webkit.org/show_bug.cgi?id=269310

Reviewed by Sammy Gill.

After 274325@main, API::FindClient::didFindString() is called in the completion handler of the final
FindString message if a match was found inside a frame. For PDFs, this frame ID was never set, so
didFindString would never be called. For now, if a match is found in a PDF, mark the main frame as the
frame containing the match.

* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):

Canonical link: https://commits.webkit.org/274585@main
